src/Illuminate/Events/Dispatcher.php

```diff
@@ -156,19 +156,31 @@ public function flush($event)
      *
      * @param  object|string  $subscriber
      * @return void
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
      */
     public function subscribe($subscriber)
     {
         $subscriber = $this->resolveSubscriber($subscriber);
 
-        $subscriber->subscribe($this);
+        $events = $subscriber->subscribe($this);
+
+        if (is_array($events)) {
+            foreach ($events as $event => $listeners) {
+                foreach (array_unique($listeners) as $listener) {
```

Comment:
👎 for the array_unique

Reply:
https://github.com/laravel/framework/blob/192356b8d519dc307bd453786551fc5c765c7cab/src/Illuminate/Foundation/Support/Providers/EventServiceProvider.php#L34

this is copied from, I kept it in case you want to extract the logic into a single method.

Reply:
It is in the service provider for a different reason. It is not needed here. I will remove it.